wrapstuff-doc-en.tex

```diff
@@ -0,0 +1,256 @@
+% !TeX program  = XeLaTeX
+% !TeX encoding = UTF-8
+%
+% This is file `wrapstuff-doc-en.tex`.
+% This file contains an English translation of the manual for the `wrapstuff` package (Copyright (C) 2022 by Qing Lee <sobenlee@gmail.com>).
+%
+% Translation by Jasper Habicht <mail@jasperhabicht.de>.
+% This file is under the same licence as the relevant files of the `wrapstuff` package.
+%
+\documentclass{ctxdoc}
+\usepackage{wrapstuff}
+\usepackage{graphicx}
+\usepackage{zhlipsum}
+\usepackage{lipsum}
+
+\usepackage[french,english]{babel}
+\hyphenation{rid-icu-lus}
+
+\ExplSyntaxOn
+\makeatletter
+\DeclareDocumentCommand \gitsha { m }
+  {
+    \href { https \c_colon_str //github.com/qinglee/wrapstuff/commit/#1 }
+          { rev. ~ \texttt{#1} }
+  }
+\DeclareDocumentCommand \ADDSPECEIALKEY { m }
+  {
+    \cs_gset_eq:NN \@@_codedoc_names_typeset_save:
+                   \__codedoc_names_typeset:
+    \cs_gset_protected:Npn \__codedoc_names_typeset:
+      {
+        \cs_gset_eq:NN \__codedoc_names_typeset:
+                       \@@_codedoc_names_typeset_save:
+        \__codedoc_names_typeset:
+        #1 \\
+      }
+  }
+\DeclareDocumentCommand \IMAGE { m m }
+  { \includegraphics [ width = \dim_eval:n {#1} ] { example-image-#2.pdf } }
+\tl_put_left:Nn \Example
+  { \trivlist \item \relax }
+\tl_put_right:Nn \endExample
+  { \endtrivlist }
+\SideBySideExampleSet { numbers = none }
+\makeatother
+\ExplSyntaxOff
+
+\DeclareDocumentCommand \EXAMPLE { }
+  {
+    \selectlanguage{french}
+    Frère Jacques, Frère Jacques, \par
+    Dor\-mez-vous? Dor\-mez-vous? \par
+    Sonnez les matines! Sonnez les matines! \par
+    Ding, dang, dong. Ding, dang, dong. \par
+    \selectlanguage{english}
+  }
+
+\ctexset{
+  abstractname   = Abstract,
+  indexname      = Index,
+  section/format = \Large\bfseries\raggedright,
+  section/name   = {{},.\space},
+  figurename     = Figure,
+}
+
+\begin{document}
+
+  \changes{v0.1}{2022/07/05}{Initial version.}
+  \changes{v0.1}{2022/07/20}{First version submitted to CTAN.}
+  \changes{v0.2}{2022/07/21}{改进放置方式。}
+
+  \title{\bfseries The \pkg{wrapstuff} Package}
+  \author{LI Qing\thanks{Translation by Jasper Habicht <mail@jasperhabicht.de>}}
+  \date{\filedate\qquad\fileversion\thanks{\gitsha{\ExplFileVersion}.}}

```

Comment:
The resulting package date from this line is wrong, you're missing the `\GetFileId{wrapstuff.sty}` call from line 182 to correctly set the `\filedate` and `\fileversion`.

Reply:
Should be fixed by `\GetFileId` as well (as that also calls `\GetIdInfo` which should set `\ExplFileVersion`).

Reply:
@Skillmon It does not. I am actually not sure where the commit ID comes from as I could not find it anywhere. If I add `\GetFileId`, I get `-1` as result. It seems that the `ctxdoc` package looks for a file with `.id` ending and the current job name. The package maintainer probably has this file. As temporary fix, I would just hard code the commit ID.